docs/packages/omc3/workflow.md

```diff
@@ -1,7 +1,824 @@
+To do so, `omc3` provides the `hole_in_one` entrypoint, which will perform frequency analysis of the data when provided with the `--harpy` flag.
+
+The script provides options involved in both data cleaning and parameter tweaking for the harmonic analysis, which is useful when you have relevant information about your measurements. 
+To use these, refer to the `Harpy Kwargs` section of the [hole_in_one API documentation][hole_in_one].
+
+In our example we will leave most of these to their default values to keep the analysis simple, but ask from `harpy` to output all computed results.
+We will input `lhc` for the `--tbt_datatype` flag, but if you skipped the use of the `tbt_converter` you should input the type of your machine there.
+Running the frequency analysis then goes as:
+```bash
+python -m omc3.hole_in_one --harpy \
+    --files trackone.sdds \
+    --tbt_datatype lhc \
+    --turns 0 1023 \
+    --autotunes transverse \
+    --to_write lin spectra full_spectra bpm_summary \
+    --outputdir harpy_output
+```
+
+??? warning "Memory Requirements"
+    During frequency analysis, `harpy` makes use of zero padding (with by default $2^{20}$ zeros) to improve the accuracy of results.
+    In turn, the padded arrays become consequent and, coupled with a high number of observation points, processing will require a substantial amount of RAM - well into the few GBs range for our example.
+
+    When running on limiting hardware, one can change the amount of zero padding with the `--turn_bits` flag for `harpy`.
+    It is important however to remember that decreasing this number will reduce the accuracy of the results, since it increases the range between detected frequencies.
+    Refer to the [hole_in_one API documentation][hole_in_one] for details that could help in determining which number to use.
+
+In the output directory, `harpy` will create TFS files with the results of the analysis for both good BPMs and identified bad BPMs.
+The filenames are determined by appending the appropriate suffixes to the entry files.
+
+!!! info "Bad BPMs"
+    If given `bpm_summary` for the `--to_write` flag (which is the case by default), `harpy` will output the `.bad_bpms_[xy]` files.
+    BPMs are determined as bad or not depending on several options from the cleaning phase.
+    While these are given sensible defaults, one might need to tweak them manually depending on their measurement. 
+    Additionally, known bad BPMs can be given with the `--bad_bpms` flag.
+
+In the output files, various properties are given in column form for each observation point.
+Running `ls harpy_output/` yields the following result:
+```bash
+trackone.sdds.ampsx		    trackone.sdds.freqsx
+trackone.sdds.ampsy		    trackone.sdds.freqsy
+trackone.sdds.bad_bpms_x	trackone.sdds.linx
+trackone.sdds.bad_bpms_y	trackone.sdds.liny
+```
+
+??? question "Column Nomenclature"
+    TODO: add a reference with the meaning of each column in the important files (the `.lin*` ones)
+
```

Comment:
as an extra hint, one may add that there is `omc3/plotting/plot_spectrum.py` which plots the spectrum at given BPMs or a waterfall plot for all BPMs to allow for some debugging or checks